dvc/remote/base.py

```diff
@@ -899,27 +906,36 @@ def _checkout_dir(
 
             entry_checksum_info = {self.PARAM_CHECKSUM: entry_checksum}
             if relink or self.changed(entry_info, entry_checksum_info):
+                modified = True
                 self.safe_remove(entry_info, force=force)
                 self.link(entry_cache_info, entry_info)
                 self.state.save(entry_info, entry_checksum)
             if progress_callback:
                 progress_callback(str(entry_info))
 
-        self._remove_redundant_files(path_info, dir_info, force)
+        modified = (
+            self._remove_redundant_files(path_info, dir_info, force)
+            or modified
+        )
 
         self.state.save_link(path_info)
         self.state.save(path_info, checksum)
 
+        if added or modified:
+            return not added and modified
+
```

Comment:
This section is not really clear for me and it can return `None` and `bool`
Maybe just 
```python
return modified and not added
```

Reply:
@JIoJIaJIu, thanks for the review. This is something that I will fix it later on (but, before, I'd like to find what's better in terms of product POV).

If you have any suggestions regarding how we could do this better, please do tell. :slightly_smiling_face: